src/Protocol/Connection.php

```diff
@@ -40,7 +41,7 @@ class Connection extends EventEmitter
     /**
      * @var Stream
      */
-    protected $stream;
+    public $stream;
```

Comment:
What is the reason for making `$stream` public? The tests?

Reply:
It is required to pause the connection if there are no pending tasks. This enables the loop to end without explicitly disconnecting the client. Other options: add pause/resume method to connection or expose the stream via a getter

Reply:
I'd suggest pause/resume methods. This is still a kind of leaky abstraction but better than full access to the connection object.